src/main/java/org/apache/pulsar/io/jcloud/format/JsonFormat.java

```diff
@@ -128,8 +131,12 @@ public ByteBuffer recordWriterBuf(Iterator<Record<GenericRecord>> record) throws
         return ByteBuffer.wrap(stringBuilder.toString().getBytes(StandardCharsets.UTF_8));
     }
 
-    private Map<String, Object> convertRecordToObject(GenericRecord record, Schema<?> schema) throws IOException {
+    protected Map<String, Object> convertRecordToObject(GenericRecord record, Schema<?> schema) throws IOException {
         if (record.getSchemaType().isStruct()) {
+            if (record.getSchemaType() == SchemaType.JSON && jsonIgnoreSchemaRead) {
+                JsonNode nativeObject = (JsonNode) record.getNativeObject();
+                return JSON_MAPPER.get().convertValue(nativeObject, TYPEREF);
+            }
```

Comment:
How about we just return the raw JSON data instead of adding a new configuration `jsonIgnoreSchemaRead`?
Are there any compatibility issue that we need to consider?

Reply:
Make sense to me, I will do it.